test/CommandLineUtils.Tests/MappedOptionAttributeTests.cs

```diff
@@ -0,0 +1,329 @@
+// Copyright (c) Nate McMaster.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using McMaster.Extensions.CommandLineUtils.Attributes;
+using McMaster.Extensions.CommandLineUtils.Conventions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace McMaster.Extensions.CommandLineUtils.Tests
+{
+    public class MappedOptionAttributeTests : ConventionTestBase
+    {
+        public MappedOptionAttributeTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        protected CommandLineApplication<T> Create<T>() where T : class
+            => Create<T, MappedOptionAttributeConvention>();
+
+        private class AppWithUnknownOptionType
+        {
+            [MappedOption("--blubb", "blubb")] public MappedOptionAttributeTests? Option { get; }
+        }
+
+        [Fact]
+        public void ThrowsWhenOptionTypeCannotBeDetermined()
+        {
+            var ex = Assert.Throws<InvalidOperationException>(
+                () => Create<AppWithUnknownOptionType>());
+            Assert.Equal(
+                Strings.CannotDetermineOptionType(typeof(AppWithUnknownOptionType).GetProperty("Option")),
+                ex.Message);
+        }
+
+        private class ShortNameOverride
+        {
+            [MappedOption(nameof(Detail1), ShortName = "d1")]
+            public string? Detail1 { get; set; }
+
+            [MappedOption(nameof(Detail2), ShortName = "d2")]
+            public string? Detail2 { get; set; }
+        }
+
+        [Fact]
+        public void CanOverrideShortNameOnOption()
+        {
+            var app = Create<ShortNameOverride>();
+
+            using var assertionScope = new AssertionScope();
```

Comment:
I personally think usage of AssertionScope usually indicates a test is actually doing more than one thing. As a matter of preference, I prefer to have more and smaller test methods which test a single behavior.